### DISCUSSION TITLE
XML documentation comments

### DISCUSSION THREAD

**Discussion by kdub-nz** [wtertinek/Linq2Acad#18] [Ideas]
Hi Wolfgang,
Thanks for the LINQ2ACAD Library, It seens to have a lot of potertial.

Have you given thought to adding XML documentation comments. 
(https://learn.microsoft.com/en-us/visualstudio/ide/reference/generate-xml-documentation-comments?view=vs-2022)
This would provide intellisence in the editor and also provide a transportable documentation.  ( at minimal cost ). Both these would reduce the learning curve and increase user proficiency I think.

Thanks for your effort. 

Regards,
Kerry Brown

**Comment by wtertinek** [comment]
Hi Kerry,

actually there are XML documentation comments. I just tried it out with Linq2Acad for AutoCAD 2023:

![grafik](https://user-images.githubusercontent.com/5783287/200527982-20643267-d822-4d83-9b33-3eb0e340a81f.png)

What AutoCAD version do you use? Do you include Linq2Acad via Nuget or via source code reference?

Regards,
Wolfgang

**Reply by kdub-nz** [reply]
Hi Wolfgang,
I used source from Github on ACAD 2023.
I'll try the Nuget in the morning.

Thanks,
Regards,
Kerry

**Reply by wtertinek** [reply]
I recommend the Nuget package, it's the easiest way to use and update the library.

**Reply by kdub-nz** [reply]
Wolfgang,
Yes, the Nuget package solved the Intellisense issue for me.

Now need to find a couple days for playing :)

Thanks,
Stay well.

**Reply by benkoshy** [reply]
Hey Kerry, it should work even when the source code is referenced in a project rather than nuget? 

The only thing I can think of which may cause problems is targetting a different .net framework version.

**Reply by kdub-nz** [reply]
Hi Ben, 
I target 4.8 as per the Doc recommendations. I have noticed that the L2A NuGet provides 4.7, but I can't imagine that will be an issue. 

I s'pose I'm a little spoiled and expecting the additional information that can be provided from the XML comments.
